realtek: add support for HPE 1920-8g-poe+ (65W)

Kevinjil opened the pull request:
Hardware information:
---------------------

- RTL8380 SoC
- 8 Gigabit RJ45 PoE ports (built-in RTL8218B)
- 2 SFP ports (built-in SerDes)
- RJ45 RS232 port on front panel
- 32 MiB NOR Flash
- 128 MiB DDR3 DRAM
- PT7A7514 watchdog
- PoE chip
- Fanless

Known issues:
---------------------
- PoE LEDs are uncontrolled.

(Manual taken from f2f09bc00280)
Booting initramfs image:
------------------------

- Prepare a FTP or TFTP server serving the OpenWrt initramfs image and connect the server to a switch port.

- Connect to the console port of the device and enter the extended boot menu by typing Ctrl+B when prompted.

- Choose the menu option "<3> Enter Ethernet SubMenu".

- Set network parameters via the option "<5> Modify Ethernet Parameter". Enter the FTP/TFTP filename as "Load File Name" ("Target File Name" can be left blank, it is not required for booting from RAM). Note that the configuration is saved on flash, so it only needs to be done once.

- Select "<1> Download Application Program To SDRAM And Run".

Initial installation:
---------------------

- Boot an initramfs image as described above, then use sysupgrade to install OpenWrt permanently. After initial installation, the bootloader needs to be configured to load the correct image file

- Enter the extended boot menu again and choose "<4> File Control", then select "<2> Set Application File type".

- Enter the number of the file "openwrt-kernel.bin" (should be 1), and use the option "<1> +Main" to select it as boot image.

- Choose "<0> Exit To Main Menu" and then "<1> Boot System".

NOTE: The bootloader on these devices can only boot from the VFS filesystem which normally spans most of the flash. With OpenWrt, only the first part of the firmware partition contains a valid filesystem, the rest is used for rootfs. As the bootloader does not know about this, you must not do any file operations in the bootloader, as this may corrupt the OpenWrt installation (selecting the boot image is an exception, as it only stores a flag in the bootloader data, but doesn't write to the filesystem).

Example PoE config file (/etc/config/poe):
---------------------
```
config global
        option budget   '65'

config port
        option enable   '1'
        option id       '1'
        option name     'lan8'
        option poe_plus '1'
        option priority '2'
config port
        option enable   '1'
        option id       '2'
        option name     'lan7'
        option poe_plus '1'
        option priority '2'
config port
        option enable   '1'
        option id       '3'
        option name     'lan6'
        option poe_plus '1'
        option priority '2'
config port
        option enable   '1'
        option id       '4'
        option name     'lan5'
        option poe_plus '1'
        option priority '2'
config port
        option enable   '1'
        option id       '5'
        option name     'lan4'
        option poe_plus '1'
        option priority '2'
config port
        option enable   '1'
        option id       '6'
        option name     'lan3'
        option poe_plus '1'
        option priority '2'
config port
        option enable   '1'
        option id       '7'
        option name     'lan2'
        option poe_plus '1'
        option priority '2'
config port
        option enable   '1'
        option id       '8'
        option name     'lan1'
        option poe_plus '1'
        option priority '2'
```

Kevinjil commented:
Corrected a mistake in the commit message, the example PoE config had the wrong PoE budget configured.

CHKDSK88 commented on target/linux/realtek/image/rtl838x.mk:
```diff
@@ -99,13 +99,22 @@ define Device/hpe_1920-8g
   SOC := rtl8380
   DEVICE_MODEL := 1920-8G-PoE+ (JG922A)
   H3C_DEVICE_ID := 0x00010025
 endef
```
`SUPPORTED_DEVICES += hpe_1920-8g-poe` is missing. People with OpenWrt 23.05 will have problem with the sysupgrade without it.

CHKDSK88 commented on target/linux/realtek/image/rtl838x.mk:
```diff
@@ -99,13 +99,22 @@ define Device/hpe_1920-8g
+define Device/hpe_1920-8g-poe-65w
+  $(Device/hpe_1920)
+  SOC := rtl8380
+  DEVICE_MODEL := 1920-8G-PoE+ (JG921A)
```
Maybe `DEVICE_MODEL := 1920-8G-PoE+ 65W (JG921A)` ?

CHKDSK88 commented on target/linux/realtek/image/rtl838x.mk:
```diff
@@ -99,13 +99,22 @@ define Device/hpe_1920-8g
+define Device/hpe_1920-8g-poe-180w
   $(Device/hpe_1920)
   SOC := rtl8380
   DEVICE_MODEL := 1920-8G-PoE+ (JG922A)
```
Maybe `DEVICE_MODEL := 1920-8G-PoE+ 180W (JG922A)`?

CHKDSK88 reviewed (commented):
Hi @Kevinjil 

Some nitpicks from me.

Kevinjil commented:
@CHKDSK88 Thank you for the feedback. I have implemented it in the branch.

CHKDSK88 commented:
@Kevinjil 
It's ok for me, but I'm not the core developer. Maybe @svanheule could help.

svanheule commented:
@CHKDSK88 Thanks for looking at these changes already.

In my opinion it would be preferable (though this is approaching nit-picking) to swap the order of the commits, and make the contents slightly different. These could do the following:
1. A commit that renames the old device name to the '180W' model, ensuring backwards compatibility by use of `SUPPORTED_DEVICES`. You can use a more formal [`Suggested-by:`](https://www.kernel.org/doc/html/latest/process/submitting-patches.html#using-reported-by-tested-by-reviewed-by-suggested-by-and-fixes) to acknowledge @CHKDSK88 suggestion.
2. A commit that purely adds the 65W model.

Currently you add the 65W model first, which requires you make name changes that aren't supported by an explanation in the commit message. It also creates the situation where, for one specific commit, no backwards compatible images can be created for the 180W model. If you do the rename first, the repository will be in a functional state at every single point.

Kevinjil commented:
@svanheule When rewriting the commits, is starting from the latest main branch preferred or should I keep the same parent commit as the current commits?

svanheule commented:
We don't normally perform merges, so patches have to be rebased anyway. Starting from a recent commit would be preferable, but I don't expect any issues in this case. You can just force push the updates.

Kevinjil committed:
realtek: rename hpe,1920-8g-poe to match hardware

There are two hardware models of the HPE 1920-8g-poe switch. The version
currently in the repository is the model with a PoE budget of 180W. In
preparation of the addition of the 65W model, the existing model is
renamed to clarify the hardware version it targets.

As suggested by Pawel, the 'SUPPORTED_DEVICES' includes the old target
name to enable an upgrade path of builds with the old name.

Signed-off-by: Kevin Jilissen <info@kevinjilissen.nl>
Suggested-by: Pawel Dembicki <paweldembicki@gmail.com>

Kevinjil committed:
realtek: add support for HPE 1920-8g-poe+ (65W)

Hardware information:
---------------------

- RTL8380 SoC
- 8 Gigabit RJ45 PoE ports (built-in RTL8218B)
- 2 SFP ports (built-in SerDes)
- RJ45 RS232 port on front panel
- 32 MiB NOR Flash
- 128 MiB DDR3 DRAM
- PT7A7514 watchdog
- PoE chip
- Fanless

Known issues:
---------------------
- PoE LEDs are uncontrolled.

(Manual taken from f2f09bc)
Booting initramfs image:
------------------------

- Prepare a FTP or TFTP server serving the OpenWrt initramfs image and
  connect the server to a switch port.

- Connect to the console port of the device and enter the extended
  boot menu by typing Ctrl+B when prompted.

- Choose the menu option "<3> Enter Ethernet SubMenu".

- Set network parameters via the option "<5> Modify Ethernet Parameter".
  Enter the FTP/TFTP filename as "Load File Name" ("Target File Name"
  can be left blank, it is not required for booting from RAM). Note that
  the configuration is saved on flash, so it only needs to be done once.

- Select "<1> Download Application Program To SDRAM And Run".

Initial installation:
---------------------

- Boot an initramfs image as described above, then use sysupgrade to
  install OpenWrt permanently. After initial installation, the
  bootloader needs to be configured to load the correct image file

- Enter the extended boot menu again and choose "<4> File Control",
  then select "<2> Set Application File type".

- Enter the number of the file "openwrt-kernel.bin" (should be 1), and
  use the option "<1> +Main" to select it as boot image.

- Choose "<0> Exit To Main Menu" and then "<1> Boot System".

NOTE: The bootloader on these devices can only boot from the VFS
filesystem which normally spans most of the flash. With OpenWrt, only
the first part of the firmware partition contains a valid filesystem,
the rest is used for rootfs. As the bootloader does not know about this,
you must not do any file operations in the bootloader, as this may
corrupt the OpenWrt installation (selecting the boot image is an
exception, as it only stores a flag in the bootloader data, but doesn't
write to the filesystem).

Example PoE config file (/etc/config/poe):
---------------------
config global
        option budget   '65'

config port
        option enable   '1'
        option id       '1'
        option name     'lan8'
        option poe_plus '1'
        option priority '2'
config port
        option enable   '1'
        option id       '2'
        option name     'lan7'
        option poe_plus '1'
        option priority '2'
config port
        option enable   '1'
        option id       '3'
        option name     'lan6'
        option poe_plus '1'
        option priority '2'
config port
        option enable   '1'
        option id       '4'
        option name     'lan5'
        option poe_plus '1'
        option priority '2'
config port
        option enable   '1'
        option id       '5'
        option name     'lan4'
        option poe_plus '1'
        option priority '2'
config port
        option enable   '1'
        option id       '6'
        option name     'lan3'
        option poe_plus '1'
        option priority '2'
config port
        option enable   '1'
        option id       '7'
        option name     'lan2'
        option poe_plus '1'
        option priority '2'
config port
        option enable   '1'
        option id       '8'
        option name     'lan1'
        option poe_plus '1'
        option priority '2'

Signed-off-by: Kevin Jilissen <info@kevinjilissen.nl>

Kevinjil commented:
@svanheule I have changed the commits.

svanheule commented:
Thanks!

Merged with commits:
987c96e889 realtek: rename hpe,1920-8g-poe to match hardware
f4ee08677c realtek: add support for HPE 1920-8g-poe+ (65W)

Kevinjil commented:
Thank you @svanheule :partying_face: 
Would it be possible to also cherry-pick these commits in the 23.05 branch?

svanheule commented:
I've picked the two commits and did a quick build test just to be sure. Since 23.05.0 is just out, these changes will only be included in 23.05.1
